### ISSUE TITLE
dataloading bug in transformer example

### ISSUE THREAD

**Issue by tungloong** [asyml/texar-pytorch#155]

**Comment by tungloong** [comment]
During testing, reference file follows original order, but model's output has been shuffled. (got a blue score 0.53)
```
python bleu_main.py --reference=data/en_vi/test.vi --translation=temp/test.output.spm
```

https://github.com/asyml/texar-pytorch/blob/026163887f1bca450f51a4730986a953d940050c/examples/transformer/transformer_main.py#L69-L84

Should set valid and test dataset 's hparams `{'shuffle': False}`, default is `True`

I wrote a simple but not elegant one.
```
    def update_hparams(hparmas, split):
        new_hparmas = hparmas.copy()
        new_hparmas['shuffle'] = True if split is "train" else False
        return new_hparmas

    datasets = {
        split: data_utils.Seq2SeqData(
            os.path.join(
                config_data.input_dir,
                f"{config_data.filename_prefix}{split}.npy"
            ),
            hparams=update_hparams(data_hparams, split),
            device=device
        ) for split in ["train", "valid", "test"]
    }
```

**Comment by huzecong** [comment]
Thanks for the issue! This is indeed a bug, we will address this in the next update.

**Comment by gpengzhi** [comment]
Resolved in #156
